lib/srv/db/proxyserver.go

```diff
@@ -380,6 +413,13 @@ func (s *ProxyServer) Connect(ctx context.Context, proxyCtx *common.ProxyContext
 	return nil, trace.BadParameter("failed to connect to any of the database servers")
 }
 
+// isReverseTunnelDownError returns true if the provided error indicates that
+// the reverse tunnel connection is down e.g. because the agent is down.
+func isReverseTunnelDownError(err error) bool {
+	return trace.IsConnectionProblem(err) ||
+		strings.Contains(err.Error(), reversetunnel.NoDatabaseTunnel)
```

Comment:
nit: Any reason why not to create an `var ErrNoDatabaseTunnel` and just use here `err == ErrNoDatabaseTunnel` instead of comparing strings? For me sounds cleaner and safer as current implementation panics when `err == nil` - not very likely to happen but still.

Reply:
As far as I looked at it, we just get a generic trace error from reversetunnel/transport in this case so I'm not sure introducing another error type is going to help. Otherwise, the check for trace.ConnectionProblem which we already had here would have worked too.